asteroid_filterbanks/pcen.py

```diff
@@ -0,0 +1,165 @@
+from torch import nn
+import torch
+from . import transforms
+from typing import Union, TypedDict
+
+
+class ExponentialMovingAverage(nn.Module):
+    """
+    Computes the exponential moving average of an sequential input.
+
+    Heavily inspired by leaf-audio's tensorflow implementation
+        https://github.com/google-research/leaf-audio/blob/7ead2f9fe65da14c693c566fe8259ccaaf14129d/leaf_audio/postprocessing.py#L27
+
+    See license here
+        https://github.com/google-research/leaf-audio/blob/master/LICENSE
+    """
+
+    def __init__(
+        self,
+        smooth: float = 0.04,
+        per_channel: bool = False,
+        n_channels: int = 2,
+        trainable: bool = False,
+    ):
+        super().__init__()
+        if per_channel:
+            self.weights = nn.Parameter(
+                torch.full((n_channels,), fill_value=smooth), requires_grad=trainable
+            )
+        else:
+            self.weights = nn.Parameter(
+                torch.full((1,), fill_value=smooth), requires_grad=trainable
+            )
+
+    def forward(self, mag_spec, initial_state):
+        weights = torch.clamp(self.weights, 0.0, 1.0)
+        accumulator = initial_state
+        out = None
+        for x in torch.split(mag_spec, 1, dim=-1):
+            accumulator = weights * x + (1.0 - weights) * accumulator
+            if out is None:
+                out = accumulator
+            else:
+                out = torch.cat((out, accumulator), dim=-1)
+        return out
+
+
+class TrainableParameters(TypedDict):
+    alpha: bool
+    delta: bool
+    root: bool
+    smooth: bool
+
+
+class PCEN(nn.Module):
+    """
+        Per-Channel Energy Normalization as described in [1].
+
+        PCEN is the use of an automatic gain control based dynamic compression to replace the widely used static compression.
+        Optional, the parameters can be trained.
+
+        Can
+            This applies a fixed or learnable normalization by an exponential moving
+    average smoother, and a compression.
+
+        Args:
+            alpha (float): exponent of EMA smoother
+                Defaults to 0.96
+            delta (float):
+                Defaults to 2.0
+            root (float):
+                Defualts to 2.0
+            floor (float):
+                Defaults to 1e-6
+            smooth (float):
+                Defaults to 0.04
+            n_channels (int):
+                Defaults to 2
+            trainable: (bool or )
```

Comment:
Interested if this API design for `trainable` fits with the current project's style.

Reply:
Yes, this is great!